modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/AttachmentUtil.java

```diff
@@ -370,6 +370,69 @@ public static Map<Locale, String> getTitleMap(
 		return cpAttachmentFileEntry.getTitleMap();
 	}
 
+	public static CPAttachmentFileEntry updateCPAttachmentFileEntry(
+			CPAttachmentFileEntry cpAttachmentFileEntry,
+			CPAttachmentFileEntryService cpAttachmentFileEntryService,
+			CPDefinitionOptionRelService cpDefinitionOptionRelService,
+			CPDefinitionOptionValueRelService cpDefinitionOptionValueRelService,
+			CPOptionService cpOptionService, Attachment attachment,
+			long classPK, int type, ServiceContext serviceContext)
+		throws Exception {
+
+		Calendar displayCalendar = CalendarFactoryUtil.getCalendar(
+			serviceContext.getTimeZone());
+
+		if (attachment.getDisplayDate() != null) {
+			displayCalendar = DateConfigUtil.convertDateToCalendar(
+				attachment.getDisplayDate());
+		}
+
+		DateConfig displayDateConfig = new DateConfig(displayCalendar);
+
+		Calendar expirationCalendar = CalendarFactoryUtil.getCalendar(
+			serviceContext.getTimeZone());
+
+		expirationCalendar.add(Calendar.MONTH, 1);
+
+		if (attachment.getExpirationDate() != null) {
+			expirationCalendar = DateConfigUtil.convertDateToCalendar(
+				attachment.getExpirationDate());
+		}
+
+		DateConfig expirationDateConfig = new DateConfig(expirationCalendar);
+
+		String options = _getJSON(
+			cpDefinitionOptionRelService, cpDefinitionOptionValueRelService,
+			cpOptionService, attachment.getOptions(), classPK,
+			serviceContext.getCompanyId());
+
+		if (options.isEmpty()) {
+			options = cpAttachmentFileEntry.getJson();
+		}
```

Comment:
this looks wrong you should use the cpAttachmentFileEntry.getJson() when the attachment.getOptions() is null, and you should wrap all of this in the _getJSON, I'm sorry I thought this was clear